### PR TITLE
Fix warnings on julia 0.4

### LINKED ISSUE #15: Test error on latest julia release-0.4
Here is the test error:

```
ERROR: LoadError: argument is an abstract type; size is indeterminate
 in randnans at /home/garrison/.julia/v0.4/SortingAlgorithms/test/runtests.jl:44
 [inlined code] from /home/garrison/.julia/v0.4/SortingAlgorithms/test/runtests.jl:98
```

The code in question (`reinterpret(Float64,[rand(Uint64)|0x7ff8000000000000 for i=1:100])`) works on julia-0.4.0-rc1 but fails on more recent `release-0.4` commits (e.g. JuliaLang/julia@1af871b).

### PR DESCRIPTION
**Pull request by garrison**
Also closes #15.

### CODE REVIEW

File: src/SortingAlgorithms.jl [hunk 1]
```diff
@@ -42,12 +42,12 @@ end
 
 # Map a bits-type to an unsigned int, maintaining sort order
 uint_mapping(::ForwardOrdering, x::Unsigned) = x
-for (signedty, unsignedty) in ((Int8, Uint8), (Int16, Uint16), (Int32, Uint32), (Int64, Uint64), (Int128, Uint128))
+for (signedty, unsignedty) in ((Int8, @compat UInt8), (Int16, @compat UInt16), (Int32, @compat UInt32), (Int64, @compat UInt64), (Int128, @compat UInt128))
```
Comment (yuyichao) [inline src/SortingAlgorithms.jl]: No need for the `@compat`.

Comment (garrison) [inline src/SortingAlgorithms.jl]: oh yes :)

**Comment by yuyichao** [comment]
0.3 Coverage failure should be fixed by https://github.com/JuliaLang/METADATA.jl/pull/3432 .

**Comment by garrison** [comment]
thanks!

**Pull request merged**